src/plugins/home/server/services/tutorials/lib/tutorial_schema.ts

```diff
@@ -5,121 +5,153 @@
+  title: schema.maybe(schema.string()),
+  callOut: schema.maybe(
+    schema.object({
+      title: schema.string(),
+      message: schema.maybe(schema.string()),
+      iconType: schema.maybe(schema.string()),
+    })
+  ),
   // Variants (OSes, languages, etc.) for which tutorial instructions are specified.
-  instructionVariants: Joi.array().items(instructionVariantSchema).required(),
-  statusCheck: statusCheckSchema,
+  instructionVariants: schema.arrayOf(instructionVariantSchema),
+  statusCheck: schema.maybe(statusCheckSchema),
 });
-
-const paramSchema = Joi.object({
-  defaultValue: Joi.required(),
-  id: Joi.string()
-    .regex(/^[a-zA-Z_]+$/)
-    .required(),
-  label: Joi.string().required(),
-  type: Joi.string().valid(Object.values(PARAM_TYPES)).required(),
+export type InstructionSetSchema = TypeOf<typeof instructionSetSchema>;
+
+const idRegExp = /^[a-zA-Z_]+$/;
+const paramSchema = schema.object({
+  defaultValue: schema.any(),
+  id: schema.string({
+    validate(value: string) {
+      if (!idRegExp.test(value)) {
+        return `Does not satisfy regexp ${idRegExp.toString()}`;
+      }
+    },
+  }),
+  label: schema.string(),
+  type: schema.oneOf([schema.literal('number'), schema.literal('string')]),
 });
+export type ParamsSchema = TypeOf<typeof paramSchema>;
 
-const instructionsSchema = Joi.object({
-  instructionSets: Joi.array().items(instructionSetSchema).required(),
-  params: Joi.array().items(paramSchema),
+const instructionsSchema = schema.object({
+  instructionSets: schema.arrayOf(instructionSetSchema),
+  params: schema.maybe(schema.arrayOf(paramSchema)),
 });
-
-export const tutorialSchema = {
-  id: Joi.string()
-    .regex(/^[a-zA-Z0-9-]+$/)
-    .required(),
-  category: Joi.string().valid(Object.values(TUTORIAL_CATEGORY)).required(),
-  name: Joi.string().required(),
-  moduleName: Joi.string(),
-  isBeta: Joi.boolean().default(false),
-  shortDescription: Joi.string().required(),
-  euiIconType: Joi.string(), // EUI icon type string, one of https://elastic.github.io/eui/#/icons
-  longDescription: Joi.string().required(),
-  completionTimeMinutes: Joi.number().integer(),
-  previewImagePath: Joi.string(),
-
+export type InstructionsSchema = TypeOf<typeof instructionSchema>;
+
+const tutorialIdRegExp = /^[a-zA-Z0-9-]+$/;
+export const tutorialSchema = schema.object({
+  id: schema.string({
+    validate(value: string) {
+      if (!tutorialIdRegExp.test(value)) {
+        return `Does not satisfy regexp ${tutorialIdRegExp.toString()}`;
+      }
+    },
+  }),
+  category: schema.oneOf([
+    schema.literal('logging'),
+    schema.literal('security'),
+    schema.literal('metrics'),
+    schema.literal('other'),
+  ]),
+  name: schema.string({
+    validate(value: string) {
+      if (value === '') {
+        return 'is not allowed to be empty';
+      }
+    },
+  }),
+  moduleName: schema.maybe(schema.string()),
+  isBeta: schema.maybe(schema.boolean({ defaultValue: false })),
```

Comment:
NIT: I think you don't need the `maybe` as a default value is provided?

Reply:
I'd say we don't need `defaultValue` here. As you can see, the plugin doesn't use `tutorialSchema`  output filled with `defaultValue` https://github.com/elastic/kibana/pull/100201/files/2bf83514fdf7cddf9b87e021993987def3b9bad8#diff-9cac3b32e0ce29a47722775e1340e462b3605fe321f23b08a89fd6fba10965a8R46